### PR TITLE
Migrated `objcluster` datasource to Framework

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Migrated `objcluster` datasource to Framework. Also fixed small issue in `kernel` datasource.

## ✔️ How to Test

`make PKG_NAME="linode/objcluster" testacc`
`make PKG_NAME="linode/kernel" testacc`

Ticket: TPT-1994

### CODE REVIEW

**Comment by ezilber-akamai** [comment]
/acctest sha=4ed80f8660033eb359c1796670e23fb115f3dfd4 pkg=linode/objcluster

**Comment by ezilber-akamai** [comment]
/acctest sha=4ed80f8660033eb359c1796670e23fb115f3dfd4 pkg=linode/kernel

File: linode/objcluster/framework_datasource_schema.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		},
+		"region": schema.StringAttribute{
+			Description: "The region this cluster is located in.",
+			Optional:    true,
+			Computed:    true,
+		},
+		"static_site_domain": schema.StringAttribute{
+			Description: "The base URL for this cluster used when hosting static sites.",
+			Optional:    true,
+			Computed:    true,
```
Comment (lgarber-akamai) [inline linode/objcluster/framework_datasource_schema.go:31]: These fields were marked as optional in the SDKv2 implementation of this resource, although it doesn't look like users could resolve a cluster using anything but its ID. Could these be removed?

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
Great work! Just one small thing

File: linode/objcluster/framework_datasource_schema.go (hunk 1 above)
Comment (ezilber-akamai) [reply linode/objcluster/framework_datasource_schema.go]: Fixed!

**Comment by ezilber-akamai** [comment]
/acctest sha=666c3fa2640a43d0cba1288af1f43fd221b91f11 pkg=linode/objcluster

File: linode/objcluster/framework_datasource_schema.go (hunk 1 above)
Comment (ezilber-akamai) [reply linode/objcluster/framework_datasource_schema.go]: I think I may have misunderstood. Do we wont to drop these fields entirely?

**Review by yec-akamai — APPROVED** [review]

**Review by zliang-akamai — APPROVED** [review]

**Review by lgarber-akamai — APPROVED** [review]

**Comment by ezilber-akamai** [comment]
/acctest sha=0c5aa1bf41b8b1f15e96955f4cc46fa26b66c61b pkg=linode/objcluster

**Pull request merged**